### PR TITLE
Andoid-SDK update to Java-SDK

### PR DESCRIPTION
**Pull request by titusgicheru**
- New Azure registration instructions
- MSAL Update to 0.2.2
- Graph Update to 1.2.0
- Code refactor to accommodate the above

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -59,4 +55,7 @@ dependencies {
     androidTestImplementation 'com.android.support.test.espresso:espresso-core:2.2.2'
     androidTestImplementation 'com.android.support.test.espresso:espresso-intents:2.2.2'
     androidTestImplementation 'com.android.support.test.espresso:espresso-web:2.2.2'
+    implementation 'com.microsoft.graph:microsoft-graph:1.2.0'
+    implementation 'com.microsoft.identity.client:msal:0.2.2'
```
Comment (NakulSabharwal) [inline app/build.gradle]: This is not needed nice auth is being done using android auth

File: README.md [hunk 2]
```diff
@@ -88,25 +88,23 @@ To use the Microsoft Graph SDK snippets project, you need the following:
...
+1. In the list of pages for the app, select **Manifest**, and: 
+    - In the manifest editor, set the ``allowPublicClient`` property to **true** 
+    - Select **Save** in the bar above the manifest editor. 
+1. In the list of pages for the app, select **API permissions** 
+    - Click the **Add a permission** button and then, 
+    - Ensure that the **Microsoft APIs** tab is selected 
+    - In the *Commonly used Microsoft APIs* section, click on **Microsoft Graph** 
+    - In the **Delegated permissions** section, ensure that the right permissions are checked. Use the search box if necessary.
+    - **User.Read**, **Mail.Send**, **Mail.ReadWrite**, **Files.ReadWrite**, **User.ReadBasic.All**, **Calendars.ReadWrite**, **Group.Read.All**  
+    - Select the **Add permissions** button
```
Comment (NakulSabharwal) [inline README.md:107]: > Paste the clipboard contents in **app/src/main/AndroidManifest.xml**, line 42 to replace `ENTER_YOUR_CLIENT_ID` with the clipboard contents.

Please make these steps in sync with changes. That where do we have to provide the client id.

File: app/src/main/java/com/microsoft/graph/snippets/AuthenticationManager.java [hunk 3]
```diff
@@ -30,14 +33,18 @@
     private static PublicClientApplication mPublicClientApplication;
     private AuthenticationResult mAuthResult;
     private MSALAuthenticationCallback mActivityCallback;
+    public static  MSALAuthenticationProvider msalAuthenticationProvider;
```
Comment (NakulSabharwal) [inline app/src/main/java/com/microsoft/graph/snippets/AuthenticationManager.java]: Where is this being used? You created a separate instance of msalAuthProvider in SnippetApp

File: app/src/main/java/com/microsoft/graph/snippets/application/SnippetApp.java [hunk 4]
```diff
@@ -6,34 +6,45 @@
...
         mAuthenticationManager = AuthenticationManager.getInstance();
     }
 
     public IGraphServiceClient getGraphServiceClient() {
-        IClientConfig clientConfig = DefaultClientConfig.createWithAuthenticationProvider(
-                mAuthenticationManager
-        );
-        return new GraphServiceClient.Builder().fromConfig(clientConfig).buildClient();
+
+        MSALAuthenticationProvider msalAuthenticationProvider = new MSALAuthenticationProvider(
```
Comment (NakulSabharwal) [inline app/src/main/java/com/microsoft/graph/snippets/application/SnippetApp.java]: You can store msalAuthenticationProvider in a member variable and initilize it only once when its null. Same with graphClient.

File: app/src/main/java/com/microsoft/graph/snippets/ServiceConstants.java [hunk 5]
```diff
@@ -10,5 +10,5 @@
     // The Microsoft Graph delegated permissions that you set in the application
     // registration portal must match these scope values.
     // Update this constant with the scope (permission) values for your application:
-    public static final String[] SCOPES = {"openid", "Mail.ReadWrite","mail.send","Files.ReadWrite","User.ReadBasic.All"};
+    public static final String[] SCOPES = {"openid", "Mail.ReadWrite","Mail.Send","Files.ReadWrite","User.ReadBasic.All", "Calendars.ReadWrite","Group.Read.All"};
```
Comment (NakulSabharwal) [inline app/src/main/java/com/microsoft/graph/snippets/ServiceConstants.java]: Pls make sure these scopes cover all the scenarios in this sample.

File: app/src/main/java/com/microsoft/graph/snippets/AuthenticationManager.java (hunk 3 above)
Comment (titusgicheru) [reply app/src/main/java/com/microsoft/graph/snippets/AuthenticationManager.java]: Updated

File: app/src/main/java/com/microsoft/graph/snippets/application/SnippetApp.java (hunk 4 above)
Comment (titusgicheru) [reply app/src/main/java/com/microsoft/graph/snippets/application/SnippetApp.java]: Updated

**Pull request merged**